### ISSUE TITLE
Regression in building RPM for linux-other as of f705c0aeb77c6dbd8d061d66ba4d0d33fb08b2bd

### ISSUE THREAD

**Issue by mandree** [wireapp/wire-desktop#1434]
Wire version: Git f705c0aeb77c6dbd8d061d66ba4d0d33fb08b2bd
Wire for web version: (not attained)
Operating system: Fedora Linux 27 (Workstation)
Which antivirus software do you have installed: none

What steps will reproduce the problem?
1. checkout the mentioned Git version f705c0aeb77c6dbd8d061d66ba4d0d33fb08b2bd or any subsequent up to 3533de1 (I have git bisected from tag release/3.1.2822 all through 3533de1beb5593bc774980af9db079e9e6e71ea6 on the master branch)
2. git checkout -- . # this is to clean up
3. adjust the checkout path in my script and run it

What is the expected result?
* receive an .rpm file

What is the actual result?
* since the mentioned commit either the build fails or there is no more .rpm written to any directory.

```
#!/bin/sh
set -ex
DIR="$HOME/GIT/wire-desktop"    # <---- ADJUST THIS!
cd "$DIR"
yarn install
yarn test
grunt -v --arch=x64 --target=rpm linux-other
ls -lbdtr wrap/dist/*.rpm
```

Screen log of successful build (release/3.1.2822):
```
+ grunt -v --arch=x64 --target=rpm linux-other
Initializing
Command-line options: --verbose, --arch=x64, --target=rpm

Reading "Gruntfile.js" Gruntfile...OK

Registering Gruntfile tasks.

Registering "grunt-contrib-clean" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-contrib-clean/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-contrib-clean/package.json...OK
Loading "clean.js" tasks...OK
+ clean

Registering "grunt-git" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-git/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-git/package.json...OK
Loading "git.js" tasks...OK
+ gitadd, gitapply, gitarchive, gitcheckout, gitclean, gitclone, gitcommit, gitdiff, gitfetch, gitlog, gitmerge, gitpull, gitpush, gitrebase, gitreset, gitrevParse, gitrm, gitstash, gitstatus, gittag

Registering "grunt-github-changes" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-github-changes/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-github-changes/package.json...OK
Loading "github_changes.js" tasks...OK
+ githubChanges

Registering "grunt-gitinfo" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-gitinfo/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-gitinfo/package.json...OK
Loading "gitInfo.js" tasks...OK
+ gitinfo
Reading package.json...OK
Parsing package.json...OK
Reading info.json...OK
Parsing info.json...OK
Initializing config...OK
Loading "Gruntfile.js" tasks...OK
+ bump-version, bundle, create-windows-installer, electron, electronbuilder, linux, linux-other, linux-other-internal, linux-prod, macos, macos-prod, productbuild, release-internal, release-prod, update-keys, version-inc, win, win-prod

Running tasks: linux-other

Running "linux-other" task

Running "clean:linux" (clean) task
Verifying property clean.linux exists in config...OK
Files: [no src] -> linux
Options: force=false, no-write=false
>> 0 paths cleaned.

Running "update-keys" task
Options: config="electron/js/config.js"
Reading electron/js/config.js...OK
Writing electron/js/config.js...OK

Running "gitinfo" task
config.options: cwd=null
>> local.branch.current.lastCommitAuthor = "AndyTheGiant"
>> local.branch.current.lastCommitMessage = "fix: remove storage data for removed accounts (#1063)
>> 
>> "
>> local.branch.current.lastCommitTime = "2017-10-18 11:43:40 +0200"
>> local.branch.current.currentUser = Matthias Andree
>> local.branch.current.shortSHA = d6a37b1
>> local.branch.current.SHA = d6a37b1631c16e2a170f173fb96ffe62de96f0a8
>> local.branch.current.name = HEAD
>> local.branch.current.lastCommitNumber = 623
>> remote.origin.url = https://github.com/wireapp/wire-desktop.git

Running "release-prod" task
Reading electron/package.json...OK
Parsing electron/package.json...OK
Writing electron/package.json...OK
Releases URL points to https://wire-app.wire.com/win/prod/ OK

Running "bundle" task

Running "electronbuilder:linux_other" (electronbuilder) task
Verifying property electronbuilder.linux_other exists in config...OK
File: [no files]
Options: asar=false, arch="x64", productName="wire-desktop", targets=["rpm"], linux={"fpm":["--name","wire-desktop"],"executableName":"wire-desktop","desktop":{"StartupWMClass":"Wire"}}
electron-builder 18.6.2
Rebuilding native production dependencies for linux:x64
⚠️  Packaging using asar archive is disabled — it is strongly not recommended.
Please enable asar and use asarUnpack to unpack files that must be externally available.
Packaging for linux x64 using electron 1.7.9 to wrap/dist/linux-unpacked
Building rpm
+ ls -lbdtr wrap/dist/wire-3.1.0-d6a37b1.x86_64.rpm
-rw-rw-r--. 1 mandree mandree 94254626 30. Apr 14:04 wrap/dist/wire-3.1.0-d6a37b1.x86_64.rpm
```

Screen log of failing build:
```
$ LC_ALL=C build-wire.sh 
+ DIR=/home/mandree/VCS-other/wire-desktop.git
+ cd /home/mandree/VCS-other/wire-desktop.git
+ yarn install
yarn install v1.6.0
$ cd electron && npm install
up to date in 0.724s
[1/4] Resolving packages...
success Already up-to-date.
Done in 1.83s.
+ yarn test
yarn run v1.6.0
$ npm run lint && npm run jest && electron-mocha tests

> wire-desktop@ lint /home/mandree/VCS-other/wire-desktop.git
> eslint electron


> wire-desktop@ jest /home/mandree/VCS-other/wire-desktop.git
> jest

 PASS  electron/renderer/src/lib/__tests__/verifyObjectProperties.spec.js
 PASS  electron/renderer/src/reducers/__tests__/accountReducer.spec.js
 PASS  electron/renderer/src/actions/__tests__/actions.spec.js
 PASS  electron/renderer/src/lib/__tests__/localStorage.spec.js
 PASS  electron/renderer/src/lib/__tests__/accentColor.spec.js

Test Suites: 5 passed, 5 total
Tests:       20 passed, 20 total
Snapshots:   0 total
Time:        0.813s, estimated 1s
Ran all test suites.


  cert pinning
    ✓ pins used hostnames
    ✓ does not pin other hostnames
    ✓ verifies all hostnames (302ms)

  tray
    #updateBadgeIcon()
      ✓ should update badge according to window title (312ms)


  4 passing (627ms)

Done in 5.79s.
+ grunt -v --arch=x64 --target=rpm linux-other
Initializing
Command-line options: --verbose, --arch=x64, --target=rpm

Reading "Gruntfile.js" Gruntfile...OK

Registering Gruntfile tasks.

Registering "grunt-contrib-clean" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-contrib-clean/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-contrib-clean/package.json...OK
Loading "clean.js" tasks...OK
+ clean

Registering "grunt-git" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-git/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-git/package.json...OK
Loading "git.js" tasks...OK
+ gitadd, gitapply, gitarchive, gitcheckout, gitclean, gitclone, gitcommit, gitdiff, gitfetch, gitlog, gitmerge, gitpull, gitpush, gitrebase, gitreset, gitrevParse, gitrm, gitstash, gitstatus, gittag

Registering "grunt-gitinfo" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-gitinfo/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-gitinfo/package.json...OK
Loading "gitInfo.js" tasks...OK
+ gitinfo
Reading package.json...OK
Parsing package.json...OK
Reading info.json...OK
Parsing info.json...OK
Initializing config...OK
Loading "Gruntfile.js" tasks...OK
+ bump-version, bundle, create-windows-installer, electron, electronbuilder, linux, linux-other, linux-other-internal, linux-prod, macos, macos-prod, productbuild, release-internal, release-prod, update-keys, version-inc, win, win-prod

Running tasks: linux-other

Running "linux-other" task

Running "clean:linux" (clean) task
Verifying property clean.linux exists in config...OK
Files: [no src] -> linux
Options: force=false, no-write=false
>> 0 paths cleaned.

Running "update-keys" task
Options: config="electron/js/config.js"
Reading electron/js/config.js...OK
Writing electron/js/config.js...OK

Running "gitinfo" task
config.options: cwd=null
>> local.branch.current.lastCommitAuthor = "Gregor Herdmann"
>> local.branch.current.lastCommitMessage = "fix: Remove early return in main (#1425)
>> 
>> "
>> local.branch.current.lastCommitTime = "2018-04-27 14:47:33 +0200"
>> local.branch.current.currentUser = Matthias Andree
>> local.branch.current.shortSHA = 3533de1
>> local.branch.current.SHA = 3533de1beb5593bc774980af9db079e9e6e71ea6
>> local.branch.current.name = master
>> local.branch.current.lastCommitNumber = 809
>> remote.origin.url = https://github.com/wireapp/wire-desktop.git

Running "release-prod" task
Reading electron/package.json...OK
Parsing electron/package.json...OK
Writing electron/package.json...OK
Releases URL points to https://wire-app.wire.com/win/prod/ OK

Running "bundle" task

Running "electronbuilder:linux_other" (electronbuilder) task
Verifying property electronbuilder.linux_other exists in config...OK
File: [no files]
Options: asar=false, arch="x64", productName="wire-desktop", targets=["rpm"], linux={"fpm":["--name","wire-desktop"],"executableName":"wire-desktop","desktop":{"Version":"1.1","Name":"Wire","GenericName":"The most secure collaboration platform","Categories":"Network;InstantMessaging;Chat;VideoConference","Keywords":"chat;encrypt;e2e;messenger;videocall","StartupWMClass":"Wire"}}
electron-builder 18.6.2

Done.
+ ls -lbdtr 'wrap/dist/*.rpm'
ls: cannot access 'wrap/dist/*.rpm': No such file or directory
[mandree@ryzen wire-desktop.git]$ LC_ALL=C build-wire.sh 
+ DIR=/home/mandree/VCS-other/wire-desktop.git
+ cd /home/mandree/VCS-other/wire-desktop.git
+ yarn install
yarn install v1.6.0
$ cd electron && npm install
up to date in 0.72s
[1/4] Resolving packages...
success Already up-to-date.
Done in 1.82s.
+ yarn test
yarn run v1.6.0
$ npm run lint && npm run jest && electron-mocha tests

> wire-desktop@ lint /home/mandree/VCS-other/wire-desktop.git
> eslint electron


> wire-desktop@ jest /home/mandree/VCS-other/wire-desktop.git
> jest

 PASS  electron/renderer/src/lib/__tests__/verifyObjectProperties.spec.js
 PASS  electron/renderer/src/reducers/__tests__/accountReducer.spec.js
 PASS  electron/renderer/src/actions/__tests__/actions.spec.js
 PASS  electron/renderer/src/lib/__tests__/accentColor.spec.js
 PASS  electron/renderer/src/lib/__tests__/localStorage.spec.js

Test Suites: 5 passed, 5 total
Tests:       20 passed, 20 total
Snapshots:   0 total
Time:        0.814s, estimated 1s
Ran all test suites.


  cert pinning
    ✓ pins used hostnames
    ✓ does not pin other hostnames
    ✓ verifies all hostnames (247ms)

  tray
    #updateBadgeIcon()
      ✓ should update badge according to window title (336ms)


  4 passing (599ms)

Done in 5.76s.
+ grunt -v --arch=x64 --target=rpm linux-other
Initializing
Command-line options: --verbose, --arch=x64, --target=rpm

Reading "Gruntfile.js" Gruntfile...OK

Registering Gruntfile tasks.

Registering "grunt-contrib-clean" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-contrib-clean/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-contrib-clean/package.json...OK
Loading "clean.js" tasks...OK
+ clean

Registering "grunt-git" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-git/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-git/package.json...OK
Loading "git.js" tasks...OK
+ gitadd, gitapply, gitarchive, gitcheckout, gitclean, gitclone, gitcommit, gitdiff, gitfetch, gitlog, gitmerge, gitpull, gitpush, gitrebase, gitreset, gitrevParse, gitrm, gitstash, gitstatus, gittag

Registering "grunt-gitinfo" local Npm module tasks.
Reading /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-gitinfo/package.json...OK
Parsing /home/mandree/VCS-other/wire-desktop.git/node_modules/grunt-gitinfo/package.json...OK
Loading "gitInfo.js" tasks...OK
+ gitinfo
Reading package.json...OK
Parsing package.json...OK
Reading info.json...OK
Parsing info.json...OK
Initializing config...OK
Loading "Gruntfile.js" tasks...OK
+ bump-version, bundle, create-windows-installer, electron, electronbuilder, linux, linux-other, linux-other-internal, linux-prod, macos, macos-prod, productbuild, release-internal, release-prod, update-keys, version-inc, win, win-prod

Running tasks: linux-other

Running "linux-other" task

Running "clean:linux" (clean) task
Verifying property clean.linux exists in config...OK
Files: [no src] -> linux
Options: force=false, no-write=false
>> 0 paths cleaned.

Running "update-keys" task
Options: config="electron/js/config.js"
Reading electron/js/config.js...OK
Writing electron/js/config.js...OK

Running "gitinfo" task
config.options: cwd=null
>> local.branch.current.lastCommitAuthor = "Gregor Herdmann"
>> local.branch.current.lastCommitMessage = "fix: Remove early return in main (#1425)
>> 
>> "
>> local.branch.current.lastCommitTime = "2018-04-27 14:47:33 +0200"
>> local.branch.current.currentUser = Matthias Andree
>> local.branch.current.shortSHA = 3533de1
>> local.branch.current.SHA = 3533de1beb5593bc774980af9db079e9e6e71ea6
>> local.branch.current.name = master
>> local.branch.current.lastCommitNumber = 809
>> remote.origin.url = https://github.com/wireapp/wire-desktop.git

Running "release-prod" task
Reading electron/package.json...OK
Parsing electron/package.json...OK
Writing electron/package.json...OK
Releases URL points to https://wire-app.wire.com/win/prod/ OK

Running "bundle" task

Running "electronbuilder:linux_other" (electronbuilder) task
Verifying property electronbuilder.linux_other exists in config...OK
File: [no files]
Options: asar=false, arch="x64", productName="wire-desktop", targets=["rpm"], linux={"fpm":["--name","wire-desktop"],"executableName":"wire-desktop","desktop":{"Version":"1.1","Name":"Wire","GenericName":"The most secure collaboration platform","Categories":"Network;InstantMessaging;Chat;VideoConference","Keywords":"chat;encrypt;e2e;messenger;videocall","StartupWMClass":"Wire"}}
electron-builder 18.6.2

Done.
+ ls -lbdtr 'wrap/dist/*.rpm'
ls: cannot access 'wrap/dist/*.rpm': No such file or directory
```

**Comment by mandree** [comment]
Appears fixed as of 34cd375b52d6688144cccf87d43ec930590ba7e6

**Comment by lipis** [comment]
Best kind of issues, that are being solved on their own! Thanks for the report and follow up.
